frontend/awx/resources/inventories/InventoryPage/InventoryDetails.cy.tsx

```diff
@@ -3,21 +3,84 @@ import { Inventory } from '../../../interfaces/Inventory';
 import { InventoryDetailsInner as InventoryDetails } from './InventoryDetails';
 
 describe('InventoryDetails', () => {
-  it('Component renders and displays inventory', () => {
-    cy.fixture('inventory').then((inventory: Inventory) => {
-      cy.mount(<InventoryDetails inventory={inventory} />);
-      cy.get('#name').should('have.text', 'test inventory');
-      cy.get('#type').should('have.text', 'Inventory');
-      cy.get('#organization').should('have.text', 'Default');
-      cy.get('#total-hosts').should('have.text', '1');
-      cy.get('#labels').should('have.text', 'test label');
-      cy.get('#created > .pf-v5-c-description-list__text > .date-time > .pf-v5-c-button').should(
-        'have.text',
-        'awx'
+  const kinds: Array<'' | 'smart' | 'constructed'> = ['', 'smart', 'constructed'];
+
+  kinds.forEach((kind) => {
+    it(`Component renders and displays inventory (${kind === '' ? 'regular' : kind})`, () => {
+      cy.intercept(
+        {
+          method: 'GET',
+          url: '/api/v2/inventories/*/instance_groups*',
+        },
+        {
+          fixture: 'instance_groups.json',
+        }
       );
-      cy.get(
-        '#last-modified > .pf-v5-c-description-list__text > .date-time > .pf-v5-c-button'
-      ).should('have.text', 'awx');
+      cy.fixture('inventory')
+        .then((inventory: Inventory) => {
+          inventory.kind = kind;
+
+          if (inventory.kind === 'smart') {
+            inventory.host_filter = 'name__icontains=test';
+          }
+          if (inventory.kind === 'constructed') {
+            inventory.hosts_with_active_failures = 1;
+          }
+        })
+        .then((inventory: Inventory) => {
+          cy.mount(<InventoryDetails inventory={inventory} />);
+          cy.get('#name').should('have.text', 'test inventory');
+          cy.get('#organization').should('have.text', 'Default');
+          cy.get('#total-hosts').should('have.text', '1');
+          cy.get('#labels').should('have.text', 'test label');
+          cy.get(
+            '#created > .pf-v5-c-description-list__text > .date-time > .pf-v5-c-button'
```

Comment:
Please dont use those selectors. First reasons are the updates of patternfly library, that sometimes enforces us to change things in the framework to match new classes (you can see for example v5 string which is for version 5), thus we will have to rewrite the tests too. Try to use data-cy, possibly aria label or element type (button, tr...).

Reply:
Good catch updated